libs/libxx/libcxxmini/libxx_new.cxx

```diff
@@ -69,7 +69,7 @@ FAR void *operator new(std::size_t nbytes)
     }
 #endif
 
-  DEBUGASSERT(alloc != nullptr);
```

Comment:
It is strange that `nullptr` is undefined for C++. @xiaoxiang781216 could you please provide some explanation on this?

Reply:
Since nullptr is defined in c++11, I intentionally change it to NULL to make it work with C++98.

Reply:
Maybe we can
```
#if __cplusplus < 201103L
#  define nullptr NULL
#endif
```
?
I'm not sure if that is better, but just as an option.

Reply:
is it good to introduce a macro with a simple name in a frequent used header file? I prefer to avoid the code to use the new feature if it can simply avoid in code.

Reply:
is it good to introduce a macro with a simple name in a frequent used header file? I prefer to avoid the code to use the new feature if it can simply avoid in code.

Reply:
Ok. Let's keep `NULL`, but I was not taking about adding that to a header file, but just adding it locally or maybe even passing it as `-D` to compiler that use C++98 option.

Reply:
libcxxmini is a very minimal C++ library implementation which doesn't even implement C++98 core language requirement, so why do we warry the code to use  C++ 11 new feature?